linked_list.py

```diff
@@ -0,0 +1,77 @@
+class Node(object):
+    def __init__(self, data, next_node=None):
+        self.data, self.next = data, next_node
+
+
+class LinkedList(object):
+    """Create and object-->linked_list."""
+    def __init__(self, head=None):
+        self.head = head
+
+    def insert(self, data):
+        """Put the given value into Node at head of list."""
+        inserted_node = Node(data)
+        inserted_node.next = self.head
+        self.head = inserted_node
+
+    def pop(self):
+        """Return node data, remove node from head of list."""
+        if self.head:
```

Comment:
Remember, in Python we prefer to ask forgiveness rather than permission.  Here you are checking to see if it's okay to ask for the `data` attribute of `self.head` before you try to get that value.  Don't ask permission.  Just try it and deal with the error that is raised if it fails.